src/puppetlabs/jdbc_util/pool.clj

```diff
@@ -65,12 +65,12 @@
           (loop []
             (if-let [result
                      (try
-                       (with-open [conn (.getConnection datasource)]
-                         (try (init-fn conn)
-                              (catch Exception e
-                                (swap! init-error (constantly e))
-                                (log/errorf e "%s - An error was encountered during initialization." (.getPoolName datasource))))
-                         datasource)
+                       (.close (.getConnection datasource))
```

Comment:
basically the same as https://github.com/brettwooldridge/HikariCP/blob/dev/src/main/java/com/zaxxer/hikari/pool/HikariPool.java#L514-L518